satpy/tests/modifier_tests/test_angles.py

```diff
@@ -0,0 +1,340 @@
+    def _check_cached_result(results, exp_zarr_chunks):
+        assert all(isinstance(x, xr.DataArray) for x in results)
+        # output chunks should be consistent
+        for angle_data_arr in results:
+            assert angle_data_arr.chunks == exp_zarr_chunks
+
+    @staticmethod
+    def _check_cache_and_clear(tmp_path, exp_num_zarr):
+        from satpy.modifiers.angles import _get_sensor_angles_from_sat_pos, _get_valid_lonlats
+        zarr_dirs = glob(str(tmp_path / "*.zarr"))
+        assert len(zarr_dirs) == exp_num_zarr  # two for lon/lat, one for sata, one for satz
+
+        _get_valid_lonlats.cache_clear()
+        _get_sensor_angles_from_sat_pos.cache_clear()
+        zarr_dirs = glob(str(tmp_path / "*.zarr"))
+        assert len(zarr_dirs) == 0
+
+    def test_cached_no_chunks_fails(self, tmp_path):
+        """Test that trying to pass non-dask arrays and no chunks fails."""
+        from satpy.modifiers.angles import _sanitize_args_with_chunks, cache_to_zarr_if
+
+        @cache_to_zarr_if("cache_lonlats", sanitize_args_func=_sanitize_args_with_chunks)
+        def _fake_func(data, tuple_arg, chunks):
+            return da.from_array(data)
+
+        data = list(range(5))
+        with pytest.raises(RuntimeError), \
+                satpy.config.set(cache_lonlats=True, cache_dir=str(tmp_path)):
+            _fake_func(data, (1, 2, 3), 5)
+
+    def test_cached_result_numpy_fails(self, tmp_path):
+        """Test that trying to cache with non-dask arrays fails."""
+        from satpy.modifiers.angles import _sanitize_args_with_chunks, cache_to_zarr_if
+
+        @cache_to_zarr_if("cache_lonlats", sanitize_args_func=_sanitize_args_with_chunks)
+        def _fake_func(shape, chunks):
+            return np.zeros(shape)
+
+        with pytest.raises(ValueError), \
+                satpy.config.set(cache_lonlats=True, cache_dir=str(tmp_path)):
+            _fake_func((5, 5), ((5,), (5,)))
+
+    def test_no_cache_dir_fails(self, tmp_path):
+        """Test that 'cache_dir' not being set fails."""
+        from satpy.modifiers.angles import _get_sensor_angles_from_sat_pos, get_angles
+        data = _get_angle_test_data()
+        with pytest.raises(RuntimeError), \
+                satpy.config.set(cache_lonlats=True, cache_sensor_angles=True, cache_dir=None):
+            get_angles(data)
+        with pytest.raises(RuntimeError), \
+                satpy.config.set(cache_lonlats=True, cache_sensor_angles=True, cache_dir=None):
+            _get_sensor_angles_from_sat_pos.cache_clear()
+
+    def test_solazi_correction(self):
+        """Test that solar azimuth angles are corrected into the right range."""
+        from datetime import datetime
+
+        from satpy.modifiers.angles import _get_sun_azimuth_ndarray
+
+        lats = np.array([-80, 40, 0, 40, 80])
+        lons = np.array([-80, 40, 0, 40, 80])
+
+        dt = datetime(2022, 1, 5, 12, 50, 0)
+
+        azi = _get_sun_azimuth_ndarray(lats, lons, dt)
```

Comment:
Can you please right the test against the higher level function(s) instead of this "private" one?

Reply:
the only higher level one that isn't private is `get_angles`. It's hard to get against that as it needs area definitions that would cover a bit enough region to have the whole range of angles.
So is it really necessary to test against the non-private function, especially considering most of the other tests in there are already testing private functions?
(the function above is using `_get_sensor_angles_from_sat_pos` for example)

Reply:
Fine.